lightning_server.cc

```diff
@@ -35,30 +33,18 @@ void LightningServer::start() {
 
   for (;;) {
     // Accept connection request
-    boost::asio::ip::tcp::socket socket(io_service_);
-    acceptor_.accept(socket);
+    boost::shared_ptr<boost::asio::ip::tcp::socket>
+        socket(new boost::asio::ip::tcp::socket(io_service_));
```

Comment:
Can we add a using-declaration like `using boost::asio::ip::tcp`to improve the readability of the types here?